### PR TITLE
FPLA:3253 Generate welsh language unit translation request form

### PR DESCRIPTION
**Pull request by jacmurphy**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-3253

### Change description ###

Original
Screen Shot 2021-07-23 at 15 11 06

Auto generated from `shouldGenerateTranslationFormRequestEnglishToWelsh()`
Screen Shot 2021-07-23 at 15 14 21

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/DocmosisDocumentGeneratorService.java [hunk 1]
```diff
@@ -32,6 +32,11 @@ public DocmosisDocument generateDocmosisDocument(DocmosisData templateData, Docm
         return generateDocmosisDocument(templateData.toMap(mapper), template);
     }
 
+    public DocmosisDocument generateDocmosisDocument(DocmosisData templateData, DocmosisTemplates template,
+                                                     RenderFormat format) {
+        return generateDocmosisDocument(templateData.toMap(mapper), template, format);
+    }
+
```
Comment (jacmurphy) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/DocmosisDocumentGeneratorService.java:39]: Ensure coverage is maintained

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/docmosis/TranslationRequestFormCreationServiceDocmosisTest.java [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                String[] words = line.split(" ");
+                numberOfWords = numberOfWords + words.length;
+            }
+            bufferedReader.close();
+        } catch (IOException exception) {
+           return 0;
+        }
+
+        return numberOfWords;
+    }
```
Comment (jacmurphy) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/docmosis/TranslationRequestFormCreationServiceDocmosisTest.java:221]: The countWordsInFile has not been finalised. I have to wait until the ticket has been updated to specify what exactly it is counting.

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/welshtranslation/DocmosisTranslationRequest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    private String name;
+    private String department;
+    private String contactInformation;
+    private DocmosisWelshProject project;
+    private String familyManCaseNumber;
+    private String description;
+    private DocmosisTranslateLanguages translate;
+    private DocmosisWelshLayout layout;
+    private int wordCount;
+    private LocalDate dateOfReturn;
```
Comment (jacmurphy) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/welshtranslation/DocmosisTranslationRequest.java:24]: This has not been finalised in the ticket. More work is required to generate a data that is suitable

File: service/src/main/java/uk/gov/hmcts/reform/fpl/enums/DocmosisTemplates.java [hunk 4]
```diff
@@ -14,7 +14,8 @@
     ORDER_V2("FL-PLW-GOR-ENG-00728.doc", "order"),
     EPO("FL-PLW-GOR-ENG-00228.doc", "emergency_protection_order"),
     EPO_V2("FL-PLW-GOR-ENG-00744.doc", "emergency_protection_order"),
-    COVER_DOCS("FL-PLW-LET-ENG-00232.doc", "cover_documents");
+    COVER_DOCS("FL-PLW-LET-ENG-00232.doc", "cover_documents"),
+    TRANSLATION_REQUEST("WELSH.doc", "translation_request");
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/enums/DocmosisTemplates.java:18]: I'd suggest to upload the form to sharepoint to have the document id (so can be reflected in local and rdo-docmosis for non prod environments - if not uploaded, the tests are going to fail in the pipeline)

File: service/src/main/java/uk/gov/hmcts/reform/fpl/NotifyTemplates.java [hunk 5]
```diff
@@ -84,4 +84,5 @@ private NotifyTemplates() {
     public static final String NOTICE_OF_CHANGE_FORMER_REPRESENTATIVE = "0411d52e-69eb-440c-bb1f-3059ac6b9130";
     public static final String MANAGING_ORGANISATION_REMOVED_TEMPLATE = "e3fb2174-128a-41a4-acbb-ea3ffe758689";
     public static final String ORDER_AMENDED_NOTIFICATION_TEMPLATE = "fc3db3ca-b62c-4a29-ad9a-864d2fe10b72";
+    public static final String WELSH_TRANSLATION_REQUEST_FORM = "cd33ac5d-37aa-4128-8a18-5d6466733185";
```
Comment (samuelzerozero) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/NotifyTemplates.java:87]: probably not part of this ticket

**Review by samuelzerozero — COMMENTED** [review]
Looks good till now

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/welshtranslation/DocmosisWelshProject.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.hmcts.reform.fpl.model.docmosis.welshtranslation;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+@AllArgsConstructor
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/welshtranslation/DocmosisWelshProject.java:9]: why is the all args constructor required here?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/welshtranslation/DocmosisWelshLayout.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.hmcts.reform.fpl.model.docmosis.welshtranslation;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+@AllArgsConstructor
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/docmosis/welshtranslation/DocmosisWelshLayout.java:9]: see previous?

**Review by lewisbirks — APPROVED** [review]

**Pull request merged**